src/cc_catalog_airflow/dags/util/loader/sql.py

```diff
@@ -288,3 +289,96 @@ def _delete_malformed_row_in_file(tsv_file_name, line_number):
             if index + 1 != line_number:
                 write_obj.write(line)
 
+
+def _create_temp_sub_prov_table(
+        postgres_conn_id,
+        temp_table='temp_sub_prov_table'
+):
+    """
+    Drop the temporary table if it already exists
+    """
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+    postgres.run(f'DROP TABLE IF EXISTS public.{temp_table};')
+
+    """
+    Create intermediary table for sub provider migration
+    """
+    postgres.run(
+        dedent(
+            f'''
+            CREATE TABLE public.{temp_table} (
+              {col.CREATOR_URL} character varying(2000),
+              {col.PROVIDER} character varying(80)
+            );
+            '''
+        )
+    )
+    postgres.run(
+        f'ALTER TABLE public.{temp_table} OWNER TO {DB_USER_NAME};'
+    )
+
+    """
+    Populate the intermediary table with the sub providers of interest
+    """
+    for sub_prov, user_id_set in prov.FLICKR_SUB_PROVIDERS.items():
+        for user_id in user_id_set:
+            creator_url = prov.FLICKR_PHOTO_URL_BASE + user_id
+            postgres.run(
+                dedent(
+                    f'''
+                    INSERT INTO public.{temp_table} (
+                      {col.CREATOR_URL} ,
+                      {col.PROVIDER}
+                    ) 
+                    VALUES (
+                      '{creator_url}' , 
+                      '{sub_prov}'
+                    );
+                    '''
+                )
+            )
+
+    return temp_table
+
+
+def update_sub_providers(
+        postgres_conn_id,
+        image_table=IMAGE_TABLE_NAME,
+        default_provider=prov.FLICKR_DEFAULT_PROVIDER
+):
+    """
+    Initially set all source values to the default provider value
+    """
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+    postgres.run(
+        dedent(
+            f'''
+            UPDATE {image_table}
+            SET {col.SOURCE} = '{default_provider}'
+            WHERE {col.PROVIDER} = '{default_provider}';
+            '''
+        )
+    )
```

Comment:
This is probably unnecessary at the DB level because I noticed that even though we originally didn't set the `source` value in Flickr API script, in the `_get_image` function in ImageStore, the `source` is set to the default provider value (which is 'flickr'). Therefore, the Source column is not null in the table, but already set to 'Flickr'. Maybe this is safe to have though because it makes the functionality more fail-safe (but also a bit more slow because there are two updates now)